x-pack/plugins/endpoint/public/applications/endpoint/store/hosts/action.ts

```diff
@@ -0,0 +1,41 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { HostListPagination, ServerApiError } from '../../types';
+import { EndpointResultList, EndpointMetadata } from '../../../../../common/types';
+
+interface ServerReturnedHostList {
+  type: 'serverReturnedHostList';
+  payload: EndpointResultList;
+}
+
+interface ServerReturnedHostDetails {
+  type: 'serverReturnedHostDetails';
+  payload: EndpointMetadata;
+}
+
+interface ServerFailedToReturnHostDetails {
+  type: 'serverFailedToReturnHostDetails';
+  payload: ServerApiError;
+}
+
+interface UserPaginatedHostList {
+  type: 'userPaginatedHostList';
+  payload: HostListPagination;
+}
+
+// https://github.com/elastic/endpoint-app-team/issues/273
```

Comment:
Maybe just add some words here as to why the action below is there - something like: 

"Why is `FakeActionWithNoPayload` here? see: <link to issue here>"